iroha/src/wsv.rs

```diff
@@ -1,13 +1,35 @@
 //! This module provides `WorldStateView` - in-memory representations of the current blockchain
 //! state.
 
+use async_std::sync::{Arc, RwLock, RwLockReadGuard};
 use config::Configuration;
+use dashmap::{
+    mapref::one::{Ref as DashmapRef, RefMut as DashmapRefMut},
+    DashSet,
+};
 use iroha_data_model::{domain::DomainsMap, peer::PeersIds, prelude::*};
 use iroha_error::Result;
-use iroha_structs::{HashSet, RwLock};
 
 use crate::{isi::FindError, prelude::*};
 
+/// Wraps [`RwLock`] so that only read operation is possible. Can be used to restrict access in certain fields or functions.
+#[derive(Debug, Clone)]
+pub struct ReadonlyLock<T>(Arc<RwLock<T>>);
```

Comment:
Hmm, May be it should be just RwLock, not `Arc<RwLock<T>>`?

Reply:
May be it can be named ReadOnlyView?

Reply:
Let's proceed with `ReadonlyLockView`.